Sources/NIO/EventLoopFuture.swift

```diff
@@ -518,17 +520,17 @@ extension EventLoopFuture {
     ///     - callback: Function that will receive the value of this `EventLoopFuture` and return
     ///         a new value lifted into a new `EventLoopFuture`.
     /// - returns: A future that will receive the eventual value.
-    public func map<U>(file: StaticString = #file, line: UInt = #line, _ callback: @escaping (T) -> (U)) -> EventLoopFuture<U> {
-        if U.self == T.self && U.self == Void.self {
-            whenSuccess(callback as! (T) -> Void)
+    public func map<U>(file: StaticString = #file, line: UInt = #line, _ callback: @escaping (Value) -> (U)) -> EventLoopFuture<U> {
```

Comment:
Same question here about `U`.

Reply:
thanks, renamed to `NewValue`, `OtherValue`, or `InputValue` depending on the context.